aten/src/ATen/native/Onehot.cpp

```diff
@@ -16,7 +16,8 @@ Tensor one_hot(const Tensor &self, int64_t num_classes) {
             return at::empty(shape, self.options());
         }
     }
-    if (self.is_meta()) {
+
+    if (self.key_set().has_all(DispatchKeySet(BackendComponent::MetaBit))) {
```

Comment:
does calling `is_meta()` directly break something?

Reply:
Hey, more context is here: https://github.com/pytorch/pytorch/pull/79170 and https://github.com/pytorch/pytorch/pull/78972

Reply:
Ah thanks for the link. Would it be useful if `FakeTensor` were able to treat `one_hot` as a primitive, instead of decomposing it?

That's actually something we want to enable for tensor subclasses soon (internal link: https://fb.quip.com/IlhcA0x3wXqu)

Reply:
Ah thanks for the link. Would it be useful if `FakeTensor` were able to treat `one_hot` as a primitive, instead of decomposing it?

That's actually something we want to enable for tensor subclasses soon (internal link: https://fb.quip.com/IlhcA0x3wXqu)

Reply:
Yea, that would be helpful!